src/EnrollWindowsHelloController.js

```diff
@@ -0,0 +1,150 @@
+/*!
+ * Copyright (c) 2015-2016, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+define([
+  'okta',
+  'util/FormController',
+  'util/FormType',
+  'vendor/webauthn',
+  'vendor/lib/q',
+  'views/enroll-factors/Footer',
+  'views/mfa-verify/WindowsHelloErrorMessageView'
+],
+function (Okta, FormController, FormType, webauthn, Q, Footer, WindowsHelloErrorMessageView) {
+
+  var _ = Okta._;
+
+  return FormController.extend({
+    className: 'enroll-windows-hello',
+    Model: {
+      local: {
+        __isEnrolled: 'boolean'
+      },
+
+      save: function () {
+        if (!webauthn.isAvailable()) {
+          return;
+        }
+
+        this.trigger('request');
+
+        if(this.get('__isEnrolled')){
+          return this.activate();
+        }
+
+        return this.doTransaction(function (transaction) {
+          return this.__enroll(transaction);
+        });
+      },
+
+      __enroll: function (transaction) {
+        var factor = _.findWhere(transaction.factors, {
+          factorType: 'webauthn',
+          provider: 'FIDO'
+        });
+
+        return factor.enroll();
+      },
+
+      activate: function () {
+        this.set('__isEnrolled', true);
+
+        return this.doTransaction(function (transaction) {
+          var activation = transaction.factor.activation;
+          var user = transaction.user;
+
+          var accountInfo = {
+            rpDisplayName: activation.rpDisplayName,
+            displayName: user.profile.displayName,
+            name: user.profile.login,
+            id: user.id
+          };
+          var cryptoParams = [{
+            type: 'ScopedCred',
+            algorithm: activation.algorithm
+          }];
+          var attestChallenge = activation.nonce;
+
+          var makeCredential = Q.defer();
+
+          webauthn.makeCredential(accountInfo, cryptoParams, attestChallenge)
```

Comment:
This looks like it's already returning a promise, no need to create another "makeCredential" promise. Just use this one to start the promise chain.